providers/gcp/resources/common.go

```diff
@@ -41,7 +41,10 @@ type assetIdentifier struct {
 }
 
 func getAssetIdentifier(runtime *plugin.Runtime) *assetIdentifier {
-	conn := runtime.Connection.(*connection.GcpConnection)
+	conn, ok := runtime.Connection.(*connection.GcpConnection)
+	if !ok {
+		return nil
```

Comment:
Should we extend this func to also return an `error`?

Reply:
There are already code paths in the same func that return nil (when no matching platform exists) so I assume returning nil here would be ok too